apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/InsertOptions.cs

```diff
@@ -30,12 +30,48 @@ public sealed class InsertOptions
         /// </summary>
         public bool? AllowUnknownFields { get; set; }
 
+        /// <summary>
+        /// Insert all valid rows of a request, even if invalid rows exist. The
+        /// default value is false, which causes the entire request to fail if any invalid
+        /// rows exist.
+        /// Use in combination with <see cref="SuppressInsertErrors"/> to obtain
+        /// information about the rows that were skipped.
+        /// </summary>
+        public bool? SkipInvalidRows { get; set; }
+
+        /// <summary>
+        /// If <see cref="SkipInvalidRows"/> is set to <c>true</c>, then you can use
```

Comment:
Assuming you agree about simplifying the behaviour, we can simplify this description too - it should just say that insert errors don't lead to an exception. You could always put that more detailed information will be available in a future release if you want.

(I'm thinking that in the future, the only change will be to return the InsertAllRowsResponse - we should still throw by default, even when SkipInvalidRows is true, to avoid it being a tricky breaking change. And users may want to set this even if they're not skipping rows, to perform their own handling.)

Reply:
Done updating the description.

And really agree with the idea of keeping this for the future as you described.